src/effprint.ml

```diff
@@ -46,6 +46,17 @@ let pp_type ?(effannot = false) ppf etype =
     let rec self ppf = function
       | Option e -> Format.fprintf ppf "%a@ option" self e
       | Ref e -> Format.fprintf ppf "%a@ ref" self e
+      | Tuple t_lst ->
+        let pp_inner_type ppf = function
+          | Tuple _ as t -> Format.fprintf ppf "@(%a@)" self t
+          | other -> self ppf other
+        in
+        let rec pp_tuple ppf = function
+          | [] -> ()
+          | [ t ] -> Format.fprintf ppf "%a@" pp_inner_type t
+          | t :: ts -> Format.fprintf ppf "%a * %a@" pp_inner_type t pp_tuple ts
+        in
+        Format.fprintf ppf "%a@" pp_tuple t_lst
```

Comment:
The nice way to print tuple types would be to add a new level between the general `pp_type` above and the more specific `pp_param_type` below:

```ocaml
and pp_tuple_type ppf t =
  let below = pp_param_type in
  match t with
  | Tuple t_lst ->
     ... (* use Format.pp_print_list, calling "below" on the types of t_lst *)
  | other -> below ppf other
```